erpnext/public/js/utils.js

```diff
@@ -694,6 +694,96 @@ $(document).on('app_ready', function() {
 		});
 	}
 
+	$.each(["Purchase Receipt", "Stock Entry", "Delivery Note"], function(i, doctype) {
+		frappe.ui.form.on(doctype, {
+			refresh: function(frm) {
+				if(!frm.doc.is_return && frm.doc.status!="Closed") {
+					if (frm.doc.docstatus == 0) {
+						frm.add_custom_button(__('Quality Inspection'), function () {
+							let data = [];
+							let inspection_type_map = {
+								"Purchase Receipt": "Incoming",
+								"Delivery Note": "Outgoing",
+								"Stock Entry": "In Process"
+							}
+							let inspection_type = inspection_type_map[frm.doc.doctype];
+							data = frm.doc.items.map(row => ({
+								"docname": row.name,
+								"reference_type": row.parenttype,
+								"reference_name": row.parent,
+								"item_code": row.item_code,
+								"item_name": row.item_name,
+								"inspection_type": inspection_type
+							}))
+							const dialog = new frappe.ui.Dialog({
+								title: __("Make Quality Inspection"),
+								fields: [
+									{
+										label: __("Items"),
+										fieldname: "items",
+										fieldtype: "Table",
+										cannot_add_rows: true,
+										data: data,
+										in_place_edit: true,
+										get_data: () => {
+											return data;
+										},
+										fields: [
+											{
+												label: __("Item Code"),
+												fieldtype: 'Link',
+												fieldname: "item_code",
+												options: "Item",
+												read_only: 1,
+												in_list_view: 1
+											},
+											{
+												label: __("Item Name"),
+												fieldtype: 'Data',
+												fieldname: "item_name",
+												read_only: 1,
+												in_list_view: 1
+											},
+											{
+												label: __("Sample Size"),
+												fieldtype: 'Float',
+												fieldname: 'sample_size',
+												in_list_view: 1
+											}
+										],
+									}
+								],
+								primary_action: function () {
+									const values = dialog.get_values().items;
+									frappe.call({
+										method: "erpnext.stock.doctype.quality_inspection.quality_inspection.make_quality_inspections",
+										freeze: true,
+										args: {
+											"items": values
+										},
+										callback: function(r) {
+											let items = r.message;
+											items.forEach(item => {
+												frappe.show_alert({
+													indicator: 'green',
+													message: __(`The Quality Inspection ${item.name} has been created `)
+												});
```

Comment:
I think we should show these in a proper dialog, since these alerts go away after 5 seconds, and the user might miss out on them